src/native/pprof.c

```diff
@@ -455,31 +495,40 @@ void pprof_durationUpdate(DProf *dp) {
       dp, (tv.tv_sec * 1000 * 1000 + tv.tv_usec) * 1000 - dp->pprof.time_nanos);
 }
 
-char pprof_Init(DProf *dp, const char **sample_names, const char **sample_units,
-                size_t n_sampletypes) {
+DProf *pprof_Init(DProf *dp, const char **sample_names,
```

Comment:
Why does this still take a `DProf *` as a parameter? I think I'm missing something.

Reply:
It's been updated to match the documentation (haa....), which is that passing a NULL pointer in the first position causes the library to heap-allocate the container and manage (free) it during pprof_free().  This is slightly inconvenient and I don't think the interface is well-suited for frequent init/free sequences since most of the inner members are heap-allocated growing arrays (which would justify the BYO-container interface), but I'd like to leave the door open a while longer for caller-provided containers, unless there are strong feelings otherwise.